### PR TITLE
Read in EDGE-T input data via madrat-style functions.

### PR DESCRIPTION
**Pull request by Loisel**
This is part of a larger effort to move the input data read-in to madrat.

### CODE REVIEW

File: R/convertTRACCS.R [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#' \dontrun{
+#' a <- readSource("TRACCS")
+#' }
+#' @author Alois Dirnaichner
+#' @seealso \code{\link{readSource}}
+#' @importFrom madrat toolCountryFill getIsoList toolCountry2isocode
+convertTRACCS <- function(x, subtype) {
+  getItems(x, dim=1) <- toolCountry2isocode(getItems(x, dim=1))
+  getSets(x)["d1.1"] <- "iso"
+  x <- toolCountryFill(x, fill = NA, countrylist = getISOlist(), verbosity = 2)
```
Comment (pfuehrlich-pik) [inline R/convertTRACCS.R:16]: Setting `countrylist = getISOlist()` might be redundant

File: R/readTRACCS.R [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+#' a <- readSource("TRACCS")
+#' }
+#' @author Alois Dirnaichner
+#' @seealso \code{\link{readSource}}
+#' @importFrom readxl read_xlsx
+#' @importFrom data.table as.data.table
+#' @importFrom magclass setComment
+#' @importFrom stringr str_extract
+readTRACCS <- function(subtype = c("roadFeDemand", "fuelEnDensity", "loadFactor", "annualMileage",
+                                   "roadVkmDemand", "roadTkmDemand", "roadPkmDemand", "railFeDemand")) {
```
Comment (pfuehrlich-pik) [inline R/readTRACCS.R:18]: I really like listing all available subtypes as you did here, maybe you could do the same for all EDGE-T related functions with subtypes?

**Review by pfuehrlich-pik — COMMENTED** [review]
Looks good! Please try to avoid NSE though :slightly_smiling_face:

File: R/convertTRACCS.R (hunk 1 above)
Comment (Loisel) [reply R/convertTRACCS.R:16]: as far as I can tell, `countrylist=NULL` in https://github.com/pik-piam/madrat/blob/master/R/toolCountryFill.R#L43

File: R/readTRACCS.R (hunk 2 above)
Comment (Loisel) [reply R/readTRACCS.R:18]: sounds reasonable. As soon as we internally agree on the structure and the available subtypes we should do that.

File: R/convertTRACCS.R (hunk 1 above)
Comment (Loisel) [reply R/convertTRACCS.R:16]:
```suggestion
  x <- toolCountryFill(x, fill = NA, verbosity = 2)
```

File: R/readTRACCS.R [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  `.` <- iso <- period <- categoryTRACCS <- vehicleType <- technology <- value <- NULL
```
[the remark this answers is not in the document]
Comment (Loisel) [reply R/readTRACCS.R:19]: I'm afraid with data.table we are not able to adhere to this policy. The central update-by-reference tool `:=` in data.table relies on NSE, and I have not found a workaround. If there is a possibility I'm happy to hear about it. These lists can be found throughout EDGE-T and if we do not want to rewrite the whole thing in dplyr we have to stick to them.

File: R/readTRACCS.R [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+      data <- rbindlist(lapply(
+        countries,
+        function(x) {
+          conv <- suppressMessages(data.table(read_excel(
+            path = file.path(
+              "TRACCS_ROAD_Final_EXCEL_2013-12-20",
+              paste0("Road Data ", x, "_Output.xlsx")),
+            sheet = "INFO","A32:B38")))
+          colnames(conv) <- c("TRACCS_technology", "cf")
+          conv[, cf := as.numeric(str_extract(cf, "\\d+\\.\\d+"))]
```
[the remark this answers is not in the document]
Comment (Loisel) [reply R/readTRACCS.R:39]: Indeed, in cases where there is no grouping, replacing columns via the list element syntax is possible. But (1) this covers only parts of the use cases of `:=`, and (2) RHS creates a copy of the vector which can be avoided with `:=`. I'm not sure if it makes sense to update *some* of the NSE which is used in EDGE-T with the downside of reduced readability (if the syntax is used elsewhere).

File: R/toolAdjustData.R [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+#' @param subtype the input parameter type
+#' @return a quitte object
+#'
+#' @importFrom rmndt magpie2dt
+#' @importFrom data.table fread
+
+toolAdjustData <- function(qobj, subtype) {
+  ## Comment: Maybe this function has to be split in toolAdjustIntensity etc.
+  ## since the dis/aggregation of vehicle classes etc might be quite lengthy
+  dt <- as.data.table(qobj)
```
[the remark this answers is not in the document]
Comment (Loisel) [reply R/toolAdjustData.R:14]: once the function is spelled out we will rely on the quitte structure of these objects. So we actually require a quitte object here. We can of course still call it `x`.

**Review by johannah-pik — APPROVED** [review]

**Pull request merged**